docs/quickstart.rst

```diff
@@ -129,6 +129,38 @@ certain number of old archives::
     borg prune -v $REPOSITORY --prefix '{hostname}-' \
         --keep-daily=7 --keep-weekly=4 --keep-monthly=6
 
+Pitfalls with shell variables and environment variables
+-------------------------------------------------------
+
+This applies to all environment variables you want borg to see, not just
+``BORG_PASSPHRASE``. The short explanation is: always ``export`` your variable,
+and use single quotes if you're unsure of the details of your shell's expansion
+behavior. E.g.::
+
+    export BORG_PASSPHRASE='your complicated & passphrase';
+
+This ensures the borg PID that needs to access this passphrase can do so. Note
+that this is not the same thing as::
+
+  $ BORG_PASSPHRASE='your complicated & passphrase'; # missing "export"!
+  $ ./yourborgjob.sh # will never see the above $BORG_PASSPHRASE value
```

Comment:
Maybe rather explain that one needs "export" to have shell variable being exported into the environment of subprocesses - so the borg process can use them. Shell variables are only visible within the script, but not to subprocesses.